com.unity.render-pipelines.core/Runtime/Utilities/CoreRenderPipelinePreferences.cs

```diff
@@ -10,80 +15,54 @@ namespace UnityEngine.Rendering
-            {
-                guiHandler = searchContext =>
-                {
-                    if (!m_Loaded)
-                        Load();
-
-                    Rect r = EditorGUILayout.GetControlRect();
-                    r.xMin = 10;
-                    EditorGUIUtility.labelWidth = 251;
-                    volumeGizmoColor = EditorGUI.ColorField(r, "Volume Gizmo Color", volumeGizmoColor);
-
-                    if (GUILayout.Button(Styles.userDefaults, GUILayout.Width(120)))
-                    {
-                        RevertColors();
-                    }
-                }
-            };
-        }
+        static readonly Color kPreviewCameraBackgroundColorDefault = new Color(82f / 255.0f, 82f / 255.0f, 82.0f / 255.0f, 0.0f);
+        public static Color previewBackgroundColor => kPreviewCameraBackgroundColorDefault;
 
-        static void RevertColors()
-        {
-            volumeGizmoColor = s_VolumeGizmoColorDefault;
-        }
+        #endregion
 
         static CoreRenderPipelinePreferences()
         {
-            Load();
+            GetColorPrefVolumeGizmoColor = RegisterSceneColor("Volume Gizmo", s_VolumeGizmoColorDefault);
         }
 
-        static void Load()
+        static readonly ConcurrentStack<object> s_ColorPref = new ConcurrentStack<object>();
+        internal static Func<Color> RegisterSceneColor(string name, Color defaultColor)
         {
-            s_VolumeGizmoColor = GetColor(Keys.volumeGizmoColor, new Color(0.2f, 0.8f, 0.1f, 0.5f));
-            m_Loaded = true;
-        }
+            if (string.IsNullOrEmpty(name))
+                throw new ArgumentException("You must give a valid name for the color property", nameof(name));
 
-        static Color GetColor(string key, Color defaultValue)
-        {
-            int value = EditorPrefs.GetInt(key, (int)ColorUtils.ToHex(defaultValue));
-            return ColorUtils.ToRGBA((uint)value);
+            // PrefColor is the type to use to have a Color that is customizable inside the Preference/Colors panel.
+            // Sadly it is internal so we must create it and grab color from it by reflection.
+            Type prefColorType = typeof(Editor).Assembly.GetType("UnityEditor.PrefColor");
+
+            var colorPref = Activator.CreateInstance(prefColorType, $"Scene/{name}", defaultColor.r, defaultColor.g, defaultColor.b, defaultColor.a);
+
+            PropertyInfo colorInfo = prefColorType.GetProperty("Color");
+            if (colorInfo == null)
+            {
+                Debug.LogError("Unable to find property `Color` on object `UnityEditor.PrefColor`");
+                return () => defaultColor;
+            }
+
+            s_ColorPref.Push(colorPref);
+
+            MemberExpression colorProperty = Expression.Property(Expression.Constant(colorPref, prefColorType), colorInfo);
+
+            return Expression.Lambda<Func<Color>>(colorProperty).Compile();
```

Comment:
It appears I forgot a case in the initial implementation:
If we load a package that use this while the Preference > Color is open, I am not sur it will redraw.

So make sure you test this. And if it not redraws, that means you must explicitly check if it is open and redraw it after each ColorPref insertion. Ask me if you don't see how to do this.